File: 백준/Silver/1026. 보물/보물.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int N = input.nextInt();
        int[] arr = new int[N];
        int[] arr2 = new int[N];

        for(int i = 0; i < N; i++){
            arr[i] = input.nextInt();
        }
        for(int i = 0; i < N; i++){
            arr2[i] = input.nextInt();
        }

        Arrays.sort(arr);
        Arrays.sort(arr2);

        int result = 0;
        for(int i = 0; i < arr.length; i++){
            result += (arr[i] * arr2[arr.length - 1 - i]);
        }
        System.out.println(result);
    }
}

File: 백준/Silver/1026. 보물/보물.kt
fun main() {
    val N = readln().toInt()
    val arr = readln().split(" ").map { it.toInt() }
    val arr2 = readln().split(" ").map { it.toInt() }
    val sortedArr = arr.sorted()
    val sortedArr2 = arr2.sorted()
    var result = 0
    for (i in arr.indices) {
        result += sortedArr[i] * sortedArr2[sortedArr.size - 1 - i]
    }
    println(result)
}
